Uses Positional Face Button Layout instead of Button Labels

AL2009man opened the pull request:
this SDL Hint will disable `SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS`. allowing all Controllers to use the positional layout (SDL3-style) as opposed to Button Labels. refer to https://github.com/fgsfdsfgs/perfect_dark/pull/693 for more info

This addresses a specific problem with Nintendo Switch controllers where the Face Buttons are assigned in a literal Button Symbol position, causes muscle memory issues while maintaining consistency across all modern controllers.

for those who...somehow, prefer the prior setup: you can restore it by either rebinding it from Key Binding (per controller) or set `useNintendoLayout` to 1 (global, affects all Controllers).

AL2009man committed:
changed Face Button Labels to Positional

AL2009man committed:
added missing variable

AL2009man committed:
Uses Positional Layout instead of Button Labels

this SDL Hint will disable `SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS`. allowing all Controllers to use the positional layout (SDL3-style) as opposed to Button Labels.

This addresses a specific problem with Nintendo Switch controllers where the Face Buttons are assigned in a literal Button Symbol position, causes muscle memory issues while maintaining consistency across all modern controllers.

for those who...somehow, prefer the prior setup: you can restore it by either rebinding it from Key Binding (per controller) or set `useNintendoLayout` to 1 (global, affects all Controllers).

fgsfdsfgs commented on port/src/input.c:
```diff
@@ -709,6 +710,18 @@ s32 inputInit(void)
+    // this is default hint in SDL 2.0.6 and later
+		SDL_SetHint(SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS, "1"); // uses button labels (A/B/X/Y)
+	} else {
+#if SDL_VERSION_ATLEAST(2, 0, 6)
```
I'd say it's not worth adding checks for `SDL_VERSION`s below 2.0.9, they're pretty ancient, so you can remove this one.

AL2009man committed:
removed `SDL_VERSION` from Postional Layout

AL2009man committed:
default N64 Bindings fix

To compensate for Positional Layout change: the `n64joybinds` for `CK_A` and `CK_B` are reversed by default, if the user happens to use a third-party N64 Controller with PC Support: this should avoid less friction

AL2009man reviewed (commented):
In case of third-party N64 Gamepads that relies on Nintendo Switch mode: the default bindings for A and B are swapped for consistency.

This workaround might not be needed at some point.

AL2009man committed:
Revert "default N64 Bindings fix"

This reverts commit 42667bd758a3bdfa907b23bdd7b8f2ca9de0dbc6.

AL2009man committed:
rename `useNintendoLayout` to `usePositionalFaceButtonLayout`

To better reflect `SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS`: the name has been changed. that also means the order of which has been swapped out.

AL2009man committed:
whitespace fix on `usePositionalFaceButtonLayout`

noticed some whitespace, and commit sync error, will have to fix and push

AL2009man committed:
Merge branch 'port' into PositionalLayout

AL2009man committed:
fixed spacing format
